modules/claims_api/app/controllers/claims_api/v2/application_controller.rb

```diff
@@ -96,6 +96,13 @@ def file_number_check(icn: params[:veteranId])
         if icn.present?
           sponsor = build_target_veteran(veteran_id: icn, loa: { current: 3, highest: 3 })
           @file_number = sponsor&.birls_id || sponsor&.mpi&.birls_id
+          if @file_number.nil?
+            claims_v2_logging('missing_file_number',
+                              message: 'missing_file_number on request in v2 application controller.')
+            raise ::Common::Exceptions::UnprocessableEntity.new(detail:
+            "Unable to locate Veteran's 'File Number' in Master Person Index (MPI). " \
+            'Please submit an issue at ask.va.gov or call 1-800-MyVA411 (800-698-2411) for assistance.')
+          end
```

Comment:
Thoughts on moving this block to the end of the method, i.e., out of the `if icn.present?` block?

We are setting `@file_number` in the below conditional, and I imagine we want our check to cover that one too.

Reply:
agreed. we could get a nil file number in the next check as well and we wouldn't alert on that, causing the same scenario. I also don't know if we'd ever get past ICN for v2 either, so that check seems a bit redundant but if we're going to have it, we should move the raise down to capture both.